### ISSUE TITLE
Unsupported data type for TPU: string in case of ControlledPQC 

### ISSUE THREAD

**Issue by Gopal-Dahale** [tensorflow/quantum#707]
Replicating the code for [Hybrid quantum-classical optimization](https://www.tensorflow.org/quantum/tutorials/hello_many_worlds#2_hybrid_quantum-classical_optimization) in the [Hello, many worlds](https://www.tensorflow.org/quantum/tutorials/hello_many_worlds) tutorial with TPU (in google cloud) results in the following error.
```
Unsupported data type for TPU: string, caused by output Pad:0 [Op:__inference_train_function_8911]
```
The modified code can be found [here](https://colab.research.google.com/drive/1IESuXimoRwKuqQJJz3UOBQqk9sc5Q1ft?usp=sharing)

**Comment by mhucka** [comment]
@Gopal-Dahale This issue is quite old, but it is not clear to me what its status is. Could you let us know, please?

**Comment by Gopal-Dahale** [comment]
@mhucka Yeah, it's so old that I forgot what it was. I think strings aren't supported with TPUs. Feel free to close this issue if it isn't a priority.

**Comment by mhucka** [comment]
Thanks for your reply.
